### DISCUSSION TITLE
VipsBlob*: vips_area_unref vs g_object_unref

### DISCUSSION THREAD

**Discussion by VivitionDeveloper** [libvips/libvips#4127] [Q&A] [answered]
I use the C++ LibVips library. I call img.dzsave_buffer() which returns a VipsBlob*. What should I do when I'm done with the blob?
1. vips_area_unref(&blob->area);
2. g_object_unref(blob);
3. Both.

Couldn't find this in the documentation.

**Comment by jcupitt** [comment, marked as answer]
Hi @CoachRDeveloper,

You can use `.write_to_buffer()`, eg:

```C++
VImage image = ...;
void *buf;
size_t size;
image.write_to_buffer(".dz", &buf, &size);
```

Passing any save options you need in the `VOption` param.

If you really want to call `.dzsave_buffer()` directly, you can see how the C++ binding calls it here:

https://github.com/libvips/libvips/blob/master/cplusplus/VImage.cpp#L748-L757

**Reply by VivitionDeveloper** [reply]
Thanks. So I guess `vips_area_unref(VIPS_AREA(blob));` is the way to go. But good to know that I can also just use `.write_to_buffer()`.
